src/Data/Repositories/WalletRepository.cs

```diff
@@ -253,5 +269,121 @@ public async Task<(bool, string?)> FinaliseWallet(Wallet selectedWalletToFinalis
+                    WalletAddressType = WalletAddressType.NativeSegwit,
+                    IsHotWallet = true, //For now, imported wallet are hot wallet that do not require user interaction
+                    IsBIP39Imported = true,
+                    BIP39Seedphrase = Constants.ENABLE_REMOTE_SIGNER ? null : seedphrase,
+                    Keys = new List<Key>()
+                };
+
+                //Persist wallet
+                var addResult = await AddAsync(wallet);
+                if (addResult.Item1 == false)
+                {
+                    _logger.LogError("Error while importing wallet from seedphrase: {Error}", addResult.Item2);
+                    return (false, addResult.Item2);
+                }
+
+                //Create key 
+                var key = new Key
+                {
+                    CreationDatetime = DateTimeOffset.Now,
+                    UpdateDatetime = DateTimeOffset.Now,
+                    Name = $"{masterFingerprint} key",
+                    XPUB = xpub,
+                    Description = "Imported BIP39 wallet key",
+                    IsArchived = false,
+                    IsCompromised = false,
+                    MasterFingerprint = masterFingerprint,
+                    Path = keyPath.ToString(),
+                    UserId = userId,
+                    IsBIP39ImportedKey = true,
+                };
+
+                //Persist key
+                var addKeyResult = await _keyRepository.AddAsync(key);
+                if (addKeyResult.Item1 == false)
+                {
+                    _logger.LogError("Error while importing wallet from seedphrase, invalid key: {Error}",
+                        addKeyResult.Item2);
+                    return (false, addKeyResult.Item2);
+                }
+
+                //Add key to wallet
+                wallet.Keys.Add(key);
+
+                //Update wallet
+                var updateResult = Update(wallet);
+                if (updateResult.Item1 == false)
+                {
+                    _logger.LogError("Error while importing wallet from seedphrase, invalid wallet: {Error}",
+                        updateResult.Item2);
+                    return (false, updateResult.Item2);
+                }
+
+                var derivationStrategyBase = wallet.GetDerivationStrategy();
+                if (derivationStrategyBase == null)
+                {
+                    return (false, "Error while getting the derivation scheme");
+                }
+
+                //Track wallet
+                await _nbXplorerService.TrackAsync(derivationStrategyBase, default);
+
+                //Since already existing wallet's utxos are not tracked by NBXplorer, we need to rescan the UTXO set for this wallet
+                //This is a long running operation in nbxplorer and should be queried in the background
+                await _nbXplorerService.ScanUTXOSetAsync(derivationStrategyBase, 1000, 30000, null, default);
```

Comment:
If this is a long running operation, should we create a job for this? Because the await is going to make it stop for a while

Reply:
The long running operation is on nbxplorer, not our side @markettes

Reply:
Perfect, thanks!